streams/src/main/java/org/apache/kafka/streams/state/internals/AbstractRocksDBSegmentedBytesStore.java

```diff
@@ -233,6 +233,7 @@ void restoreAllInternal(final Collection<KeyValue<byte[], byte[]>> records) {
                 final S segment = entry.getKey();
                 final WriteBatch batch = entry.getValue();
                 segment.write(batch);
+                batch.close();
```

Comment:
Can we add a test this this?

Reply:
Not really sure what a good way to test this would be...thoughts?

Reply:
I would be tricky. We would need to intercept the call to `new WriteBatch` within `getWriteBatches()` using power-mock and later verify if `close()` was called on the mock. Might not be worth it...